File: ts/import-content-items/importing_create_type.ts
// Tip: Find more about JS/TS SDKs at https://kontent.ai/learn/javascript
import { ManagementClient } from '@kontent-ai/management-sdk';

const client = new ManagementClient({
    environmentId: 'KONTENT_AI_ENVIRONMENT_ID',
    apiKey: 'KONTENT_AI_MANAGEMENT_API_KEY'
});

const response = await client
    .addContentType()
    .withData((builder) => {
        return {
            codename: 'cafe',
            name: 'Cafe',
            external_id: 'cafe',
            elements: [
                builder.numberElement({
                    name: 'Price per unit',
                    type: 'number',
                    codename: 'price_per_unit'
                }),
                builder.guidelinesElement({
                    guidelines:
                        '<h2>Keep Guidelines where the creative process happens.</h2>\n<p>These are sample guidelines that you can place for the whole content item. It’s a place where you can include your content brief, voice and tone recommendations or the URL to a wireframe, so the author will have all the relevant instructions at hand before writing a single line.</p>\n<p>Besides overview guidelines, you can include instructions for each particular content element, as you will see below.</p>',
                    type: 'guidelines',
                    codename: 'n2f836bce_e062_b2cd_5265_f5c3be3aa6f5'
                }),
                builder.textElement({
                    name: 'Street',
                    type: 'text',
                    codename: 'street'
                }),

                builder.textElement({
                    name: 'City',
                    type: 'text',
                    codename: 'city'
                }),
                builder.textElement({
                    name: 'Country',
                    type: 'text',
                    codename: 'country'
                }),
                builder.textElement({
                    name: 'State',
                    type: 'text',
                    codename: 'state'
                }),
                builder.textElement({
                    name: 'ZIP Code',
                    type: 'text',
                    codename: 'zip_code'
                }),
                builder.textElement({
                    name: 'Phone',
                    type: 'text',
                    codename: 'phone'
                }),
                builder.textElement({
                    name: 'Email',
                    type: 'text',
                    codename: 'email'
                }),
                builder.assetElement({
                    name: 'Photo',
                    type: 'asset',
                    codename: 'photo'
                })
            ]
        };
    })
    .toPromise();

File: net/import-content-items/importing_create_type.cs
// Tip: Find more about .NET SDKs at https://kontent.ai/learn/net
using Kontent.Ai.Management;
using Kontent.Ai.Management.Configuration;
using Kontent.Ai.Management.Models.Types;
using Kontent.Ai.Management.Models.Types.Elements;

// Or register it through DI with services.AddManagementClient()
using var client = new ManagementClient(new ManagementOptions
{
    ApiKey = "KONTENT_AI_MANAGEMENT_API_KEY",
    EnvironmentId = "KONTENT_AI_ENVIRONMENT_ID"
});

var response = (await client.CreateContentTypeAsync(new ContentTypeCreateModel
{
    Codename = "cafe",
    Name = "Cafe",
    ExternalId = "cafe",
    Elements =
    [
        new NumberElementMetadataModel
        {
            Name = "Price per unit",
            Codename = "price_per_unit",
        },
        new GuidelinesElementMetadataModel
        {
            Guidelines = "<h2>Keep Guidelines where the creative process happens.</h2>\n<p>These are sample guidelines that you can place for the whole content item. It’s a place where you can include your content brief, voice and tone recommendations or the URL to a wireframe, so the author will have all the relevant instructions at hand before writing a single line.</p>\n<p>Besides overview guidelines, you can include instructions for each particular content element, as you will see below.</p>",
            Codename = "n2f836bce_e062_b2cd_5265_f5c3be3aa6f5",
        },
        new TextElementMetadataModel
        {
            Name = "Street",
            ExternalId = "street",
        },
        new TextElementMetadataModel
        {
            Name = "City",
            ExternalId = "city",
        },
        new TextElementMetadataModel
        {
            Name = "Country",
            ExternalId = "country",
        },
        new TextElementMetadataModel
        {
            Name = "State",
            ExternalId = "state",
        },
        new TextElementMetadataModel
        {
            Name = "ZIP code",
            ExternalId = "zip_code",
        },
        new TextElementMetadataModel
        {
            Name = "Email",
            ExternalId = "email",
        },
        new TextElementMetadataModel
        {
            Name = "Phone",
            ExternalId = "phone",
        },
        new AssetElementMetadataModel
        {
            Name = "Photo",
            Codename = "photo"
        }
    ]
})).EnsureSuccess();
